osbs/cli/main.py

```diff
@@ -291,6 +295,50 @@ def cmd_resume_builds(args, osbs):
     osbs.resume_builds(namespace=args.namespace)
 
 
+def cmd_backup(args, osbs):
+    dirname = time.strftime("osbs-backup-{0}-{1}-%Y-%m-%d-%H%M%S"
+                            .format(args.instance, args.namespace))
+    if args.filename == '-':
+        outfile = sys.stdout.buffer if PY3 else sys.stdout
+    elif args.filename:
+        outfile = args.filename
+    else:
+        outfile = dirname + ".tar.bz2"
+
+    with paused_builds(osbs, args.namespace):
+        with TarWriter(outfile, dirname) as t:
+            for resource_type in BACKUP_RESOURCES:
+                logger.info("dumping %s" % resource_type)
+                resources = osbs.dump_resource(resource_type, namespace=args.namespace)
+                t.write_file(resource_type + ".json", json.dumps(resources).encode('ascii'))
+
+    if not hasattr(outfile, "write"):
+        logger.info("backup archive created: %s", outfile)
+
+
+def cmd_restore(args, osbs):
+    if args.BACKUP_ARCHIVE == '-':
+        infile = sys.stdin.buffer if PY3 else sys.stdin
+    else:
+        infile = args.BACKUP_ARCHIVE
+    asciireader = codecs.getreader('ascii')
+
+    with paused_builds(osbs, args.namespace):
+        for f in TarReader(infile):
+            resource_type = os.path.basename(f.filename).split('.')[0]
+            if resource_type not in BACKUP_RESOURCES:
+                logger.warning("Unknown resource type for %s, skipping", f.filename)
+                continue
+
+            logger.info("restoring %s" % resource_type)
```

Comment:
Silly nitpick: this, and also line 311, could be written `logger.info("... %s", resource_type)` (no `%`).